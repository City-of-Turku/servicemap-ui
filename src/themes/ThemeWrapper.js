import { ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { useSelector } from 'react-redux';
import themes from '.';
import config from '../../config';
import { selectThemeMode } from '../redux/selectors/user';

// If external theme (by Turku) is true, then can be used to select which theme to get
const externalTheme = config.themePKG;
const isExternalTheme = !externalTheme || externalTheme === 'undefined' ? null : externalTheme;

/**
 * Returns default theme based on env value
 * @returns theme
 */
const getTheme = () => {
  if (isExternalTheme) {
    return themes.SMThemeTku;
  }
  return themes.SMTheme;
};

// Component to handle theme changes
const ThemeHandler = ({ children }) => {
  const themeMode = useSelector(selectThemeMode);
  return (// Get correct theme setting from store
    <ThemeProvider theme={themeMode === 'dark' ? themes.SMThemeDark : getTheme()}>
      {children}
    </ThemeProvider>
  );
};
ThemeHandler.propTypes = {
  children: PropTypes.node,
};
ThemeHandler.defaultProps = {
  children: null,
};

export default ThemeHandler;
