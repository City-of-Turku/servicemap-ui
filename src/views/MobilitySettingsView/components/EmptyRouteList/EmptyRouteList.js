import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

/**
   * Check if route list is empty and render correct text
   * @property {any} classes
   * @property {any} intl
   * @property {Array} route
   * @return {JSX Element}
   */

const EmptyRouteList = ({ classes, intl, route }) => {
  if (route) {
    return (
      <div className={classes.paragraph}>
        <Typography
          component="p"
          variant="subtitle2"
          aria-label={
                route.length > 0
                  ? intl.formatMessage({ id: 'mobilityPlatform.menu.routes.info' })
                  : intl.formatMessage({ id: 'mobilityPlatform.menu.routes.emptyList' })
              }
        >
          {route.length > 0
            ? intl.formatMessage({ id: 'mobilityPlatform.menu.routes.info' })
            : intl.formatMessage({ id: 'mobilityPlatform.menu.routes.emptyList' })}
        </Typography>
      </div>
    );
  }
  return null;
};

EmptyRouteList.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  route: PropTypes.arrayOf(PropTypes.any),
};

EmptyRouteList.defaultProps = {
  route: [],
};

export default EmptyRouteList;
