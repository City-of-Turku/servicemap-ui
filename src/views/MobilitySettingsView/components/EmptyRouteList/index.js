import EmptyRouteList from './EmptyRouteList';

export default EmptyRouteList;
