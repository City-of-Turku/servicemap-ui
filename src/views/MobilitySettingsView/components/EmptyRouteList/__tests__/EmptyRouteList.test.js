/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../../themes';
import EmptyRouteList from '../index';
import finnishTranslations from '../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  route: [],
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <MuiThemeProvider theme={themes.SMTheme}>{children}</MuiThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<CityBikeInfo />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<EmptyRouteList {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<EmptyRouteList {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual(
      'Reittejä ei löytynyt.',
    );
  });

  it('does contain aria-label attributes', () => {
    const { container } = renderWithProviders(<EmptyRouteList {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toEqual(
      'Reittejä ei löytynyt.',
    );
  });
});
