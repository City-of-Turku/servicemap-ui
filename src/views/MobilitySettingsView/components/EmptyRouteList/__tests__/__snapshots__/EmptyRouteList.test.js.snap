// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityBikeInfo /> should work 1`] = `
<div>
  <div
    class="injectIntl(EmptyRouteList)-paragraph-1"
  >
    <p
      aria-label="Reittejä ei löytynyt."
      class="MuiTypography-root MuiTypography-subtitle2"
    >
      Reittejä ei löytynyt.
    </p>
  </div>
</div>
`;
